util/hint/hint_processor.go

```diff
@@ -63,7 +63,12 @@ func ExtractTableHintsFromStmtNode(node ast.Node) []*ast.TableOptimizerHint {
 		return x.TableHints
 	case *ast.DeleteStmt:
 		return x.TableHints
-	// TODO: support hint for InsertStmt
+	case *ast.InsertStmt:
+		// If the Select of Insert have hints, it would replace the Insert hints
```

Comment:
Does MySQL have this policy?

Reply:
In [Mysql 8.0 ducoument](https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html)

> Duplicate hints: For a hint such as /*+ MRR(idx1) MRR(idx1) */, MySQL uses the first hint and issues a warning about the duplicate hint.

And I also test with mysql 8.0, It seems the hint in the select with a higher priority.

```sh
mysql> insert /*+ qb_name(qb1) */ into a select /*+ qb_name(qb2) */ 1;
Query OK, 1 row affected, 1 warning (0.08 sec)
Records: 1  Duplicates: 0  Warnings: 1

mysql> show warnings;
+---------+------+----------------------------------------------------------+
| Level   | Code | Message                                                  |
+---------+------+----------------------------------------------------------+
| Warning | 3126 | Hint QB_NAME(`qb1`) is ignored as conflicting/duplicated |
+---------+------+----------------------------------------------------------+
1 row in set (0.00 sec)

```

Reply:
But this change may result in unexpected behaviors. For example, w/ MySQL8.0
```mysql
mysql> insert into tt select /*+ max_execution_time(1000) */ sleep(3);
Query OK, 1 row affected, 1 warning (3.01 sec)
Records: 1  Duplicates: 0  Warnings: 1

mysql> show warnings;
+---------+------+-------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                             |
+---------+------+-------------------------------------------------------------------------------------+
| Warning | 3125 | MAX_EXECUTION_TIME hint is supported by top-level standalone SELECT statements only |
+---------+------+-------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

Reply:
You're right. What about directly return insert's hints here?

Reply:
Yes, I think just let `qb_name(qb1)` affect with this SQL `insert /*+ qb_name(qb1) */ into a select /*+ qb_name(qb2) */ 1;` is ok.

Reply:
updated.